### ISSUE TITLE
Date Picker keyboard navigation test fails when run in the Paris timezone

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter#60370]
(cd packages/flutter; export TZ='Europe/Paris'; flutter test --enable-experiment=non-nullable test/material/date_picker_test.dart)

Will fail with:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: DateTime:<2016-01-18 00:00:00.000>
  Actual: DateTime:<2016-01-17 00:00:00.000>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Users/darrenaustin/dev/flutter/packages/flutter/test/material/date_picker_test.dart:1236:9)
<asynchronous suspension>
```

**Comment by darrenaustin** [comment]
Not sure what is causing the issue yet, but if I can't resolve it, I will look into locking down the timezone for this test.

**Comment by darrenaustin** [comment]
@a14n This should be fixed with #59586. Let me know if you still see any issue after picking it up.
